### PR TITLE
Rename String prefix/suffix/infix functions

### LINKED ISSUE #1174: Add String.contains
Are we really missing this function? :)

### PR DESCRIPTION
**Pull request by esbenbjerre**
Fixes #1174.
Switches the argument order of `String.contains` (formerly named `isInfixOf`) to ensure that the UFCS makes sense.

### CODE REVIEW

File: main/test/ca/uwaterloo/flix/library/TestString.flix [hunk 1]
```diff
@@ -2601,100 +2575,118 @@ def indexOfRight08(): Bool = String.indexOfRight("ab", "ab") == Some(0)
 def indexOfRight09(): Bool = String.indexOfRight("bc", "abc") == Some(1)
 
 /////////////////////////////////////////////////////////////////////////////
-// isPrefixOf                                                              //
+// startsWith                                                              //
```
Comment (magnus-madsen) [inline main/test/ca/uwaterloo/flix/library/TestString.flix:2578]: Can you check if there are already test cases for startsWith and endsWith? There might be

Comment (magnus-madsen) [reply main/test/ca/uwaterloo/flix/library/TestString.flix:2578]: Or did we just happen to have startsWith and endsWith implemented, but without any tests?

Comment (magnus-madsen) [reply main/test/ca/uwaterloo/flix/library/TestString.flix:2578]: If that is the case, we should of course keep the tests :)

Comment (esbenbjerre) [reply main/test/ca/uwaterloo/flix/library/TestString.flix:2578]: We had 3 test cases each for `startsWith` and `endsWith`. I "merged" those with the test cases for `isPrefixOf`/`isSuffixOf` (which have been removed).
I can see the git diff might look a bit weird due to my copy paste and argument order switch, but the process was as follows:
- We had the functions: `startsWith`, `endsWith`, `isPrefixOf`, `isSuffixOf`, `isInfixOf`
- I removed the `isPrefixOf` and `isSuffixOf` functions
- I renamed `isInfixOf` to `contains`
- I merged the test cases for `isPrefixOf`/`startsWith` and `isSuffixOf`/`endsWith`
- I renamed the test cases for `isInfixOf` to `contains`

Comment (magnus-madsen) [reply main/test/ca/uwaterloo/flix/library/TestString.flix:2578]: Excellent

**Pull request merged**